### PR TITLE
Feature/atoum

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,15 +13,17 @@ before_script:
   - sed -e '/"php":/d' -i composer.json
   - rm -f composer.lock
   - composer install --optimize-autoloader
+  - patch -d vendor/guzzlehttp/guzzle/ -p1 < tools/guzzle-php72.patch
```
Comment (orthagh) [inline .travis.yml]: maybe a comment about upstream (and a todo)

File: tests/APIBaseClass.php [hunk 2]
```diff
@@ -191,91 +204,102 @@ public function testGetMultipleItems($session_token) {
...
+         ->hasKey('sort')
+         ->hasKey('order')
+         ->hasKey('rawdata');
+
       $headers = $data['headers'];
-      $this->assertArrayHasKey('Accept-Range', $headers);
-      $this->assertContains('User', $headers['Accept-Range'][0]);
 
-      $this->assertNotEquals(false, $data);
-      $this->assertArrayHasKey('totalcount', $data);
```
Comment (orthagh) [inline tests/APIBaseClass.php]: you removed a lot of tests here, no ?

File: tests/APIBaseClass.php [hunk 3]
```diff
@@ -394,47 +429,57 @@ public function testSearchWithBadCriteria($session_token) {
...
+         ->isNotFalse();
+
+      $this->array($data)
+         ->hasKey('id')
+         ->hasKey('message');
+
       $computers_id = $data['id'];
-      $this->assertEquals(true, is_numeric($computers_id));
-      $this->assertEquals(true, $computers_id > 0);
-      $this->assertArrayHasKey('message', $data);
```
Comment (orthagh) [inline tests/APIBaseClass.php]: test removed

File: tests/APIBaseClass.php (hunk 2 above)
Comment (trasher) [inline tests/APIBaseClass.php]: Nope, it has been moved a few lines above in the array() test

File: tests/APIBaseClass.php (hunk 3 above)
Comment (trasher) [inline tests/APIBaseClass.php]: Nope, it has been moved a few lines above in the array() test

File: tests/APIBaseClass.php (hunk 2 above)
Comment (trasher) [inline tests/APIBaseClass.php]: They've been moved mainly

**Pull request merged**